packages/odyssey-react-mui/src/web-component/shadow-dom.ts

```diff
@@ -10,25 +10,35 @@
  * See the License for the specific language governing permissions and limitations under the License.
  */
 
-export const createShadowDomElements = (containerElement: HTMLElement) => {
-  const shadowRoot = containerElement.attachShadow({ mode: "open" });
-
-  // Container for Emotion `<style>` elements.
+export const createUnattachedShadowDomElements = () => {
+  const appRootElement = document.createElement("div");
   const emotionRootElement = document.createElement("div");
+
+  // This `div` may cause layout issues unless it inherits the parent's height.
+  appRootElement.style.setProperty("height", "inherit");
+
+  appRootElement.setAttribute("id", "app-root");
   emotionRootElement.setAttribute("id", "style-root");
   emotionRootElement.setAttribute("nonce", window.cspNonce);
 
-  // React app root component.
-  const shadowRootElement = document.createElement("div");
-  shadowRootElement.setAttribute("id", "shadow-root");
-  // This div could cause issues with layout of children.
-  // For flexibility, make it inherit its parent's height
-  shadowRootElement.style.setProperty("height", "inherit");
+  return { appRootElement, emotionRootElement };
+};
+
+export type ShadowDomElements = ReturnType<
+  typeof createUnattachedShadowDomElements
+>;
+
+export const createShadowDomElements = (containerElement: HTMLElement) => {
+  const shadowRoot = containerElement.attachShadow({ mode: "open" });
+
+  // Container for Emotion `<style>` elements.
+  const { appRootElement, emotionRootElement } =
+    createUnattachedShadowDomElements();
 
+  shadowRoot.appendChild(appRootElement);
   shadowRoot.appendChild(emotionRootElement);
-  shadowRoot.appendChild(shadowRootElement);
 
-  return { emotionRootElement, shadowRootElement };
+  return { emotionRootElement, shadowRootElement: appRootElement };
 };
 
 /** @deprecated Use `createShadowDomElements` instead. */
```

Comment:
Is this comment still accurate? 

Could we add some doc comments for the two functions above to help describe when each should be used?

Reply:
## New function: `createShadowDomElements`
Correct. I made a new `createShadowDomElements` because the older functions API was wonky and not intuitive to use. I want this function to be self-documenting, and the tuple export could be easily broken by accidentally swapping the variable names.

To ensure I didn't break existing apps, I made a new one. My original thought was we were going to use this as I made this change prior to working on web components.

Looking at it now, I'd prefer just exporting `appRootElement` rather than renaming it to `shadowRootElement`. It's named this way to match the props passed to `<OdysseyProvider>` and other functions, but I could easily change those to match. `ShadowRoot` is actually the parent node of these `div`s.

## **tl;dr** Probably not necessary
Neither of these functions (`createShadowDomElements` nor `createShadowRootElement`) will be usable when wrapped in UI Shell. That's because UI Shell comes with its own `ShadowRoot`, and you can't `ShadowRoot` be a child of another `ShadowRoot`. Folks using this function will have to use `renderReactInWebComponent` instead.

This was one of the first things I did when working on UI Shell when I thought bare Shadow DOMs would work. I had to switch to a web compnoent, so it's all moot now. The new `createShadowDomElements` isn't even used, so I'll remove it.

Reply:
I think my point in this comment is that this has existed in a published version of Odyssey, so it's now an API that's available.  Removing or renaming would be an API change. If it's an available API that folks can use, it'd be nice to describe the intent of it. (and, in fact, regardless of UI Shell, folks could/should use this separately to isolate their UI package impl from the global scope).

Reply:
`createShadowDomElements` was never published. IT was created for this PR as part of some work I was already intending to do. But I'm going to nix that in favor of `renderReactInWebComponent`.

Reply:
but it is already available, right? https://github.com/okta/odyssey/blob/main/packages/odyssey-react-mui/src/createShadowDomElements.ts#L13

Reply:
My bad. I guess I must've added it in the past. I've re-added this function and instead refactored the code around it:
3eda78d7171106eef8a61038a5f99e91cabe3e53.